### DISCUSSION TITLE
User Management -> Sync tool Configuration

### DISCUSSION THREAD

**Discussion by Banakshi-01** [thoughtspot/cs_tools#175] [General]
What should be the file format in terms of what columns should be included if i am trying to sync using excel.
Do i have maintain the file name as standard "ts_auth_sync_users" ? 

My end goal is to remove users in Bulk.
My excel list has the list of usernames i wish to keep

Attaching a screenshot for reference.
![CStools_syncerror](https://github.com/user-attachments/assets/f8d68491-460e-4302-9960-b5ce30f3f940)
![File_userlist](https://github.com/user-attachments/assets/a0cde60e-cf15-4e37-b750-ba99838475d2)

**Comment by boonhapus** [comment]
Hey @Banakshi-01 , I'm about to write a tool specifically for deleting users. The tool you're using here is wrapping [**the user/sync API**](https://developers.thoughtspot.com/docs/user-api#user-sync) , whose purpose is to replace the users and groups settings in ThoughtSpot from somewhere external (LDAP, a database table, etc). As part of that API, ThoughtSpot allows you to delete users.

.. but it's a bit of a shotgun approach. And shotguns are scary tools.

I've got some time blocked off on Friday US hours to write a user delete tool, which will take a list of users to remove from the platform.

We can also use the Searchable tool to extract the current users (along with a bunch of extra data). If you want to see this data now, you can use the following command.

`cs_tools tools searchable metadata --syncer csv://directory=. --config yourConfigName`

Your users are in a file called `ts_user.csv`.

**Comment by Banakshi-01** [comment]
Hey Team,

That sounds great.
Looking forward to the delete utility tool in that case. 😊




Internal Use - Confidential

**Reply by Banakshi-01** [reply]
Hi @boonhapus : Have you had any luck with the utility tool to be able to delete bulk user in TS ?

**Reply by boonhapus** [reply]
Hey @Banakshi-01 , [**this was released**](https://thoughtspot.github.io/cs_tools/tools/user-management/#__tabbed_1_3) in 1.5.11 !

All you need is at least 1 column which the name `username` as input. So if you ran `searchable` with a csv syncer and kept only the users you want to delete, then you can use that as input to this command.

---

`cs_tools tools searchable metadata --syncer csv://directory=.`

Open the `ts_user.csv` file and REMOVE all users you want to KEEP.

`cs_tools tools user-management delete --syncer csv://directory=. --deletion ts_user`
